pyomo/repn/plugins/mps.py

```diff
@@ -584,10 +586,16 @@ def yield_all_constraints():
             if len(col_entries) > 0:
                 var_label = variable_symbol_dictionary[id(vardata)]
                 for i, (row_label, coef) in enumerate(col_entries):
-                    output_file.write(column_template
+                    if i == 0 and skip_objective_sense and not is_min:
```

Comment:
I don't think this would be okay to merge as the MPS writer is used by other solvers than do handle 'max' objectives, and this would lead to confusing output in the solver log. This also doesn't take into account that there is a separate section for quadratic objective terms, which doesn't seem to have any coefficients changed. Perhaps more importantly, this kind of change would likely unexpectedly flip the sign of the constraint duals in the solution, which could lead to unexpected behavior in scripts that use that info.

In the short term, it's probably okay to just raise an exception when a maximization model is encountered in the MIPCL solver plugin. You might have to override one of the methods on the SystemCallSolver or OptSolver base classes to get access to the model before it goes to the MPS writer.